### PR TITLE
Fixes 3189: change gpg key file endpoint

### PR DESCRIPTION
**Pull request by rverdile**
## Summary
Changes the gpg key file endpoint to `repository_gpg_key/:uuid` so that it can be whitelisted i.e. skip auth
## Testing steps
1. Deploy PR in ephemeral
2. Create repository with GPG key and snapshot enabled
3. Use the GPG Key URL in the config.repo file to open a tab (in browser) to the GPG Key
4. Should be able to open the URL in your browser despite not having the correct auth

## Checklist

- [x] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3189

**Comment by jlsherrill** [comment]
this still didn't seem to work in ephemeral: 

curl -v https://ee-9n5k9jye.apps.c-rh-c-eph.8p0c.p1.openshiftapps.com/api/content-sources/v1.0/repositories/be1c2024-f3de-42d3-bedd-4aeb7d4a1d17/gpg_key/

HTTP/1.1 403 Forbidden

**Comment by jlsherrill** [comment]
I'm getting 'Bad Request: missing x-rh-identity header'  I suspect our auth skipper middleware needs to be updated :)

**Comment by jlsherrill** [comment]
also the URL in the config.repo file needs to be updated

File: pkg/handler/repositories.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func RegisterRepositoryRoutes(engine *echo.Group, daoReg *dao.DaoRegistry,
 	addRoute(engine, http.MethodPost, "/repositories/", rh.createRepository, rbac.RbacVerbWrite)
 	addRoute(engine, http.MethodPost, "/repositories/bulk_create/", rh.bulkCreateRepositories, rbac.RbacVerbWrite)
 	addRoute(engine, http.MethodPost, "/repositories/:uuid/introspect/", rh.introspect, rbac.RbacVerbWrite)
-	addRoute(engine, http.MethodGet, "/repositories/:uuid/gpg_key/", rh.getGpgKeyFile, rbac.RbacVerbRead)
+	addRoute(engine, http.MethodGet, "/repository_gpg_key/:uuid", rh.getGpgKeyFile, rbac.RbacVerbRead)
```
Comment (jlsherrill) [inline pkg/handler/repositories.go:56]:
```suggestion
	addRoute(engine, http.MethodGet, "/repository_gpg_key/:uuid/", rh.getGpgKeyFile, rbac.RbacVerbRead)
```

Comment (rverdile) [reply pkg/handler/repositories.go:56]: we don't usually put a trailing slash on the endpoints that end with a uuid

Comment (rverdile) [reply pkg/handler/repositories.go:56]: i mean maybe we should put a trailing slash for everything to be consistent, but we haven't

Comment (jlsherrill) [reply pkg/handler/repositories.go:56]: yes, i'm okay without one, but then all the other instances here need to be updated to not have a trailing slash

Comment (rverdile) [reply pkg/handler/repositories.go:56]: right now the other instances don't have a trailing slash. they look like this, `repositories/:uuid`

File: pkg/handler/repositories.go [hunk 2]
```diff
@@ -531,7 +531,7 @@ func (rh *RepositoryHandler) introspect(c echo.Context) error {
 // @Failure      404 {object} ce.ErrorResponse
 // @Failure      415 {object} ce.ErrorResponse
 // @Failure      500 {object} ce.ErrorResponse
-// @Router       /repositories/{uuid}/gpg_key/ [get]
+// @Router       /repository_gpg_key/{uuid}/ [get]
```
Comment (jlsherrill) [inline pkg/handler/repositories.go:534]:
```suggestion
// @Router       /repository_gpg_key/{uuid} [get]
```

File: pkg/handler/repositories.go (hunk 1 above)
Comment (jlsherrill) [reply pkg/handler/repositories.go:56]: or really just two places, commented on them :)

Comment (rverdile) [reply pkg/handler/repositories.go:56]: oh whoops

**Comment by jlsherrill** [comment]
don't forget to regen the api doc :)

File: pkg/dao/snapshots.go [hunk 3]
```diff
@@ -184,7 +184,7 @@ func (sDao *snapshotDaoImpl) GetRepositoryConfigurationFile(orgID, snapshotUUID,
 	var gpgKeyField string
 	if repoConfig.GpgKey != "" {
 		gpgCheck = 1
-		gpgKeyField = fmt.Sprintf("http://%v%v/repositories/%v/gpg_key/", host, api.FullRootPath(), repoConfigUUID) // host includes trailing slash
+		gpgKeyField = fmt.Sprintf("http://%v%v/repository_gpg_key/%v", host, api.FullRootPath(), repoConfigUUID) // host includes trailing slash
```
[the remark this answers is not in the document]
Comment (jlsherrill) [reply pkg/dao/snapshots.go:187]: although i guess defaulting to https:// wouldn't work in dev.  hrmmmmmm.  How about we change it to https for now, and i'll add the https/http issue to https://issues.redhat.com/browse/HMS-3170

[the remark this answers is not in the document]
Comment (rverdile) [reply pkg/dao/snapshots.go:187]: sure

**Review by jlsherrill — APPROVED** [review]

**Comment by swadeley** [comment]
Hi @rverdile , I put the setup steps in the Jira as its rather verbose.

I get 403 with the gpgkey URL.

**Comment by swadeley** [comment]
> Hi @rverdile , I put the setup steps in the Jira as its rather verbose.
> 
> I get 403 with the gpgkey URL.

Hi, as per @jlsherrill I ran deployment script again pointing to the local checked out PR branch to get the changes in deployment.yaml. Now it works.

**Pull request merged**